File: 2 - Add Two Numbers/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
 
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummyHead = new ListNode(0);
        int carry = 0;
        ListNode curr = dummyHead;
        
        while(l1 != null || l2 != null) {
            int v1 = (l1 != null) ? l1.val : 0;
            int v2 = (l2 != null) ? l2.val : 0;
            int sum = carry + v1 + v2;
            
            carry = sum / 10;
            curr.next = new ListNode(sum % 10);
            
            if (l1 != null) l1 = l1.next;
            if (l2 != null) l2 = l2.next;
            curr = curr.next;
        }
        
        if (carry > 0) {
            curr.next = new ListNode(carry);
        }
        
        return dummyHead.next;
    }
}

File: 2 - Add Two Numbers/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        n1 = 0
        mul = 1
        while l1 is not None:
            n1 += l1.val * mul
            mul *= 10
            l1 = l1.next
        
        n2 = 0
        mul = 1
        while l2 is not None:
            n2 += l2.val * mul
            mul *= 10
            l2 = l2.next
        
        s = n1 + n2
        
        head = ListNode(s % 10)
        s /= 10
        
        curr = head
        while s > 0:
            curr.next = ListNode(s % 10)
            s /= 10
            curr = curr.next
            
        return head